### ISSUE TITLE
Starting point to create KLV data and mux

### ISSUE THREAD

**Issue by petergerten** [paretech/klvdata#16]
I would like to use the project in the other direction:
Create KLV data and mux it with gstreamer. 
The demuxing and parsing example works great but does anyone have a good starting point to create a KLV muxed stream from scratch?

**Comment by All4Gis** [comment]
I think you should ask in the gstreamer forums.
For now we don't plan to create the multiplex tool

**Comment by paretech** [comment]
@petergerten, KLVDATA was written with this goal in mind and in its current state can be used to parse or compose data streams. As you have identified, KLVDATA is unable to mux or de-mux metadata with other streams. Reading [MISB ST 1402 "MPEG -2 Transport of Compressed Motion Imagery and Metadata"](http://www.gwg.nga.mil/misb/st_pubs.html) and ISO/IEC 13818-1 are good places to start. 

If I were implementing this, I would think that FFMPEG and gstreamer would potentially already provide the capability or require potentially minimal coding to adapt a solution.

Asynchronous metadata muxing would likely come first then extend to synchronous muxing. If I recall correctly, the primary piece of information the transport stream muxer needs needs to stitch together the two pieces is the coorelated timestamp to provide the information needed to de-muxers for presentation. MISB ST 1402 does a good job of introducing these concepts.

**Comment by petergerten** [comment]
@paretech thanks for the explanation and pointers. Much appreciated!

**Comment by All4Gis** [comment]
@petergerten I started this class, which can serve as a starting point. What I don't know how to do is create a MISB video https://github.com/All4Gis/QGISFMV/blob/master/code/manager/QgsMultiplexor.py

**Comment by wedmisten-ccri** [comment]
Are there any examples you could point to for composing KLV data? I would like to start with that before worrying about muxing the data stream back in.

**Comment by All4Gis** [comment]
Yes, this point https://github.com/All4Gis/QGISFMV/blob/master/code/manager/QgsMultiplexor.py

**Comment by wedmisten-ccri** [comment]
I was hoping there might be a simpler example or one in the klvdata library, but thank you, I will look at that.
